geo-data-server/src/main/java/org/gridsuite/geodata/server/GeoDataService.java

```diff
@@ -168,24 +186,44 @@ private void step(Step step, Network network, Map<String, Set<String>> sortedNei
         }
     }
 
-    private static SubstationGeoData calculateCentroidGeoData(Substation substation, Set<String> neighbours, Step step,
+    private static Coordinate getAverageCoordinate(List<SubstationGeoData> neighboursGeoData) {
+        double lat = neighboursGeoData.stream().mapToDouble(n -> n.getCoordinate().getLat()).average().orElseThrow(IllegalStateException::new);
+        double lon = neighboursGeoData.stream().mapToDouble(n -> n.getCoordinate().getLon()).average().orElseThrow(IllegalStateException::new);
+        return new Coordinate(lat, lon);
+    }
+
+    private SubstationGeoData calculateCentroidGeoData(Substation substation, Set<String> neighbours, Step step,
                                                               Map<String, SubstationGeoData> substationsGeoData) {
         // get neighbours geo data
         List<SubstationGeoData> neighboursGeoData = neighbours.stream().map(substationsGeoData::get)
                 .filter(Objects::nonNull)
                 .collect(Collectors.toList());
 
+        String substationCountry = substation.getNullableCountry() != null ? substation.getNullableCountry().name() : null;
         Coordinate coordinate = null;
         if (neighboursGeoData.size() > 1) {
             // centroid calculation
```

Comment:
May be remove these comment "centroid calculation", as the getAverageCoordinate call is more explicit.